### PR TITLE
Feature/SC-6060 - Analyse and fix top errors in sentry

### PR DESCRIPTION
**Pull request by DanielRieckhof**
# Description

## Links to Tickets or other pull requests

https://ticketsystem.hpi-schul-cloud.org/browse/SC-6060

## Changes
- Fix cookie settings

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -146,14 +146,16 @@ if (!Configuration.get('COOKIE__SECURE') && Configuration.get('COOKIE__SAME_SITE
 }
 
 app.use(session({
-	cookie: { maxAge: 1000 * 60 * 60 * 6 },
+	cookie: {
+		maxAge: 1000 * 60 * 60 * 6,
+		sameSite: Configuration.get('COOKIE__SAME_SITE'), // restrict jwt access to our domain ressources only
+		secure: Configuration.get('COOKIE__SECURE'),
```
Comment (tofixx) [inline app.js:152]: ...defaultOptions

File: middleware/i18n.js [hunk 2]
```diff
@@ -15,7 +16,14 @@ const middleware = async (req, res, next) => {
 
 	if (currentLanguage) {
 		changeLanguage(currentLanguage);
-		res.cookie('USER_LANG', currentLanguage);
+		res.cookie('USER_LANG', currentLanguage, {
```
Comment (tofixx) [inline middleware/i18n.js:19]: create cookie helper setCookie(res, key, value, [options]) and reuse it for jwt cookie too

Comment (tofixx) [reply middleware/i18n.js:19]: https://github.com/hpi-schul-cloud/schulcloud-client/blob/97f8e05c9b513479bfdae46efe24905835db7d2c/controllers/registration.js#L151 and https://github.com/hpi-schul-cloud/schulcloud-client/blob/0004f582b7b905dd3dede2c5ae6309d9bbdf5e9c/helpers/authentication.js#L252-L253 should use it

File: app.js [hunk 3]
```diff
@@ -146,14 +148,15 @@ if (!Configuration.get('COOKIE__SECURE') && Configuration.get('COOKIE__SAME_SITE
 }
 
 app.use(session({
-	cookie: { maxAge: 1000 * 60 * 60 * 6 },
+	cookie: {
+		...cookieDefaults,
+		maxAge: 1000 * 60 * 60 * 6,
```
Comment (tofixx) [inline app.js:153]: create a named const `SIX_HOURS = 1000 * 60 * 60 * 6`

File: middleware/i18n.js (hunk 2 above)
Comment (tofixx) [reply middleware/i18n.js:19]: here, use the helper too :)

**Review by tofixx — COMMENTED** [review]
please fix all usages of res.cookie. at least https://github.com/hpi-schul-cloud/schulcloud-client/pull/2252/files#diff-afdfbc385942aca5a06fd3e8026e50b197b4f1ac80209d949137cde3ccee6673R19 is missing

File: helpers/authentication.js [hunk 4]
```diff
@@ -249,13 +251,7 @@ const login = (payload = {}, req, res, next) => {
 	const { redirect } = payload;
 	delete payload.redirect;
 	return api(req).post('/authentication', { json: payload }).then((data) => {
-		res.cookie('jwt', data.accessToken, {
-			expires: new Date(Date.now() + Configuration.get('COOKIE__EXPIRES_SECONDS')),
```
Comment (ABialoskorski) [inline helpers/authentication.js:253]: should not be here also cookieDefaults?

**Review by ABialoskorski — APPROVED** [review]

**Review by mwolowyk-cap — APPROVED** [review]

**Pull request merged**